forma/src/segment.rs

```diff
@@ -293,7 +311,12 @@ impl SegmentBuffer {
                 (p0x, p0y, p1x, p1y)
             };
 
-            if p0y == p1y {
+            let skip = p0y == p1y
```

Comment:
Do you think it would be better to put this into a function, something like `should_skip_segment`? I see that the condition also repeats at line 499 in this file.

And possibly wrap the inner `and`s in parantheses, to make it clearer.